### PR TITLE
 PYTHON-4764 Update to use current supported EVG hosts

### PR DESCRIPTION
**Pull request by blink1073**

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -2795,7 +2795,7 @@ buildvariants:
 
 - matrix_name: "tests-windows-enterprise-auth"
   matrix_spec:
-    platform: windows-64-vsMulti-small
```
Comment (ShaneHarvey) [inline .evergreen/config.yml:2798]: windows-64-vsMulti-small is/was a special distro that was built just for us to test with. I take it we don't need that anymore?

Comment (ShaneHarvey) [reply .evergreen/config.yml:2798]: https://jira.mongodb.org/browse/BUILD-9934

Ah it was for testing python <=3.4. So yea we don't need it anymore.

Comment (blink1073) [reply .evergreen/config.yml:2798]: This is just changing the name we give it, not the actual host

File: .evergreen/config.yml [hunk 2]
```diff
@@ -2081,10 +2081,10 @@ axes:
...
       - id: rhel8
         display_name: "RHEL 8.x"
-        run_on: rhel87-small
+        run_on: rhel8.8-small
         batchtime: 10080  # 7 days
-      - id: rhel92-fips
-        display_name: "RHEL 9.2 FIPS"
+      - id: rhel9-fips
+        display_name: "RHEL 9 FIPS"
         run_on: rhel92-fips
```
Comment (Jibola) [inline .evergreen/config.yml:2088]:
```suggestion
        run_on: rhel9-fips
```

Comment (Jibola) [reply .evergreen/config.yml:2088]: Does the run_on also need to be changed?

**Review by Jibola — CHANGES_REQUESTED** [review]

File: .evergreen/config.yml (hunk 2 above)
Comment (blink1073) [reply .evergreen/config.yml:2088]: No, the host is still rhel92-fips.

**Review by ShaneHarvey — APPROVED** [review]

**Pull request merged**